integration-tests/src/tests/network/runner.rs

```diff
@@ -688,95 +596,70 @@ impl Runner {
         self.ports = Some(ports);
         self.validators = Some(validators);
 
-        let pm_addr: Vec<_> = self
-            .test_config
-            .iter()
-            .enumerate()
-            .map(|(node_id, _)| self.setup_node(node_id))
-            .collect();
-
-        RunningInfo { pm_addr, peers_info }
+        let mut nodes = vec![];
+        for (node_id, _) in self.test_config.iter().enumerate() {
+            nodes.push(Some(self.setup_node(node_id).await?));
+        }
+        Ok(RunningInfo { runner: self, nodes, peers_info })
     }
 }
 
-/// Use to start running the test.
+/// Executes the test.
 /// It will fail if it doesn't solve all actions.
-pub fn start_test(runner: Runner) {
-    run_actix(async {
-        runner.start();
-    })
-}
-
-impl Actor for Runner {
-    type Context = Context<Self>;
-
-    fn started(&mut self, ctx: &mut Self::Context) {
-        let addr = ctx.address();
-
-        init_test_logger();
-
-        let info = self.build();
-
-        let mut pointer = None;
-        let mut flag = Arc::new(AtomicBool::new(true));
-        let mut state_machine = self.state_machine.take().unwrap();
-        self.info = Some(Arc::new(RwLock::new(info)));
-
-        let info = self.info.as_ref().cloned().unwrap();
-
-        let can_write_log = Arc::new(AtomicBool::new(true));
-        WaitOrTimeoutActor::new(
-            Box::new(move |ctx| {
-                if can_write_log.swap(false, Ordering::Relaxed) == true {
-                    debug!(target: "network", "runner.rs: WaitOrTimeoutActor");
+/// start_test will block until test is complete.
+pub fn start_test(runner: Runner) -> anyhow::Result<()> {
+    init_test_logger();
+    let r = tokio::runtime::Runtime::new()?;
```

Comment:
I wonder if we should use the current-thread runtime here? From the perspetive of tests being fast and predictable, I'd say yes. From the prespective of being closer to prod, I'd say no.

Reply:
The framework code is executing synchronously anyway - and that's intended - to simplify the flow. The code under test is executing asynchronously in its own actix-managed runtime, so whether we use current-thread runtime for framework or not, doesn't affect the production-closeness. I just forgot that current-thread runtime is an option :P.